src/libtiled/tilelayer.h

```diff
@@ -513,6 +515,7 @@ class TILEDSHARED_EXPORT TileLayer : public Layer
     int mHeight;
     Cell mEmptyCell;
     QHash<QPoint, Chunk> mChunks;
+    QRegion mBounds;
```

Comment:
You only need to extend the rectangle in some cases. It's easy to do, but then, Qt even has `QRect::united`. Not going through `QRegion` will be a lot more efficient.